## Importation des différentes librairies dont nous avons besoin et ouverture d'une partie du fichier de données que nous allons étudier

In [33]:
# Import all the libraries we need for the brief
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
import re

# Reading of the TSV file by choosing 10 000 random rows
read_food_data = pd.read_csv('/home/apprenant/Documents/Brief-5-Rachid-Karbiche/data/01_raw_data/en.openfoodfacts.org.products.tsv', sep= '\t', low_memory= False)

# Showing all the rows and the columns
pd.options.display.max_columns = None
pd.options.display.max_rows = None

read_food_data.shape

(356027, 163)

Comme le fichier de données est assez volumineux, j'ai choisi de faire un premier tri en ne gardant que les entrées ayant comme valeur de countries 'France' car notre étude se concentre sur le marché français.

In [31]:
# Converting in the countries_en column all values containing 'France' into 'France'
read_food_data["countries_en"] = read_food_data["countries_en"].apply(
    lambda x: "France" if re.match(r".*(France).*", str(x), re.IGNORECASE) else x)

read_food_data = read_food_data.loc[read_food_data.countries_en == 'France']
read_food_data.shape

print

(129320, 163)

Au lieu des 356 027 entrées initiales, nous avons au final 129 320 entrées dont la valeur du pays est la France.

In [22]:
# Saving of the random rows in a new file
# random_data = read_food_data.to_csv('/home/apprenant/Documents/Brief-5-Rachid-Karbiche-2/data/01_raw_data/french_openfoodfacts_data.csv', index=False)

# Reading of the new TSV file
random_food_data = pd.read_csv('/home/apprenant/Documents/Brief-5-Rachid-Karbiche-2/data/01_raw_data/french_openfoodfacts_data.csv', low_memory= False)

## Examen des différentes colonnes de la base de données

In [23]:
print(random_food_data.shape)
print('------------------------------')
print(random_food_data.columns.values)

selected_column = random_food_data[['cities']]
selected_column.head()

print(selected_column.loc[selected_column.cities.notnull()])

(129320, 163)
------------------------------
['code' 'url' 'creator' 'created_t' 'created_datetime' 'last_modified_t'
 'last_modified_datetime' 'product_name' 'generic_name' 'quantity'
 'packaging' 'packaging_tags' 'brands' 'brands_tags' 'categories'
 'categories_tags' 'categories_en' 'origins' 'origins_tags'
 'manufacturing_places' 'manufacturing_places_tags' 'labels' 'labels_tags'
 'labels_en' 'emb_codes' 'emb_codes_tags' 'first_packaging_code_geo'
 'cities' 'cities_tags' 'purchase_places' 'stores' 'countries'
 'countries_tags' 'countries_en' 'ingredients_text' 'allergens'
 'allergens_en' 'traces' 'traces_tags' 'traces_en' 'serving_size'
 'no_nutriments' 'additives_n' 'additives' 'additives_tags' 'additives_en'
 'ingredients_from_palm_oil_n' 'ingredients_from_palm_oil'
 'ingredients_from_palm_oil_tags'
 'ingredients_that_may_be_from_palm_oil_n'
 'ingredients_that_may_be_from_palm_oil'
 'ingredients_that_may_be_from_palm_oil_tags' 'nutrition_grade_uk'
 'nutrition_grade_fr' 'pnns_group

Dans un premier temps, on constate que nous retrouvons 163 colonnes au total dans la base. Il faut donc procéder à un premier tri en enlevant les variables qui nous sont complètement inutiles dans le cadre de notre projet.

In [24]:
# selected_columns = read_food_data[['product_name', 'generic_name', 'quantity', 'brands', 'brands_tags', 'categories', 'categories_tags', 'categories_en', 'origins', 'origins_tags', 'labels', 'labels_tags', 'labels_en', 'countries', 'countries_tags', 'countries_en', 'ingredients_text', 'allergens', 'allergens_en', 'traces', 'traces_tags', 'traces_en', 'additives_n', 'additives', 'additives_tags', 'additives_en', ]]
# selected_columns.head()

Après observation de ces différentes colonnes, on constate que beaucoup sont de la métadonnée (code, url, creator, ...). Elles concernent uniquement la création de la data dans la base.

In [25]:
random_food_data.isnull().sum(axis = 0)

code                                               0
url                                                0
creator                                            1
created_t                                          0
created_datetime                                   1
last_modified_t                                    0
last_modified_datetime                             0
product_name                                    6287
generic_name                                   87722
quantity                                       47200
packaging                                      61174
packaging_tags                                 61175
brands                                         11858
brands_tags                                    11873
categories                                     52451
categories_tags                                52451
categories_en                                  52451
origins                                       111565
origins_tags                                  

In [26]:
selected_data_corr = selected_data.corr()
f,ax=plt.subplots(figsize=(10,7))
sns.heatmap(selected_data_corr, cmap='viridis')
plt.title("Correlation between features", 
          weight='bold', 
          fontsize=18)
plt.xticks(weight='bold')
plt.yticks(weight='bold')

plt.show()

NameError: name 'selected_data' is not defined